src/dodal/devices/panda_fast_grid_scan.py

```diff
@@ -0,0 +1,288 @@
+import threading
+import time
+from typing import Any
+
+import numpy as np
+from bluesky.plan_stubs import mv
+from numpy import ndarray
+from ophyd import (
+    Component,
+    Device,
+    EpicsSignal,
+    EpicsSignalRO,
+    EpicsSignalWithRBV,
+    Signal,
+)
+from ophyd.status import DeviceStatus, StatusBase
+from pydantic import BaseModel, validator
+from pydantic.dataclasses import dataclass
+
+from dodal.devices.motors import XYZLimitBundle
+from dodal.devices.status import await_value
+from dodal.parameters.experiment_parameter_base import AbstractExperimentParameterBase
+
+
+@dataclass
+class GridAxis:
+    start: float
+    step_size: float
+    full_steps: int
+
+    def steps_to_motor_position(self, steps):
+        """Gives the motor position based on steps, where steps are 0 indexed"""
+        return self.start + self.step_size * steps
+
+    @property
+    def end(self):
+        """Gives the point where the final frame is taken"""
+        # Note that full_steps is one indexed e.g. if there is one step then the end is
+        # refering to the first position
+        return self.steps_to_motor_position(self.full_steps - 1)
+
+    def is_within(self, steps):
+        return 0 <= steps <= self.full_steps
+
+
+class PandaGridScanParams(BaseModel, AbstractExperimentParameterBase):
```

Comment:
Could: A lot of these params are the same FGS one, could we get them both to inherit from something common?